File: ws/CEDENOR/U2/WS21Inheritance/Zoo/src/main/java/ec/edu/espe/model/Animal.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.model;

import java.util.Date;

/**
 *
 * @author Nahomi Cedeño, codecreators, DCCO-ESPE
 */
public abstract class Animal {
    
    private int id;
    private String description;
    private Date bornOnDate;
    private char gender;
    private Cage cage;
    private int numberOfbones;

    @Override
    public String toString() {
        return "Animal{" + "id=" + id + ", description=" + description + ", bornOnDate=" + bornOnDate + ", gender=" + gender + ", cage=" + cage + ", numberOfbones=" + numberOfbones + '}';
    }

    public Animal(int id, String description, Date bornOnDate, char gender, Cage cage, int numberOfbones) {
        this.id = id;
        this.description = description;
        this.bornOnDate = bornOnDate;
        this.gender = gender;
        this.cage = cage;
        this.numberOfbones = numberOfbones;
    }
    
    public abstract void feed(int food);
    
    public void register(Cage cage){
        System.out.println("");
    }

    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the description
     */
    public String getDescription() {
        return description;
    }

    /**
     * @param description the description to set
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * @return the bornOnDate
     */
    public Date getBornOnDate() {
        return bornOnDate;
    }

    /**
     * @param bornOnDate the bornOnDate to set
     */
    public void setBornOnDate(Date bornOnDate) {
        this.bornOnDate = bornOnDate;
    }

    /**
     * @return the gender
     */
    public char getGender() {
        return gender;
    }

    /**
     * @param gender the gender to set
     */
    public void setGender(char gender) {
        this.gender = gender;
    }

    /**
     * @return the cage
     */
    public Cage getCage() {
        return cage;
    }

    /**
     * @param cage the cage to set
     */
    public void setCage(Cage cage) {
        this.cage = cage;
    }

    /**
     * @return the numberOfbones
     */
    public int getNumberOfbones() {
        return numberOfbones;
    }

    /**
     * @param numberOfbones the numberOfbones to set
     */
    public void setNumberOfbones(int numberOfbones) {
        this.numberOfbones = numberOfbones;
    }
}

File: hw/travez/u2/HW18ZooInPython/ec_edu_espe_zoo_model/Animal.py
import sys
import os

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from abc import ABC, abstractmethod
from datetime import datetime

# Ismael Travez, The Java Bandits, DCCO-ESPE

class Animal(ABC):
    def __init__(self, id, description, date_onborn, gender, number_of_bones, cage):
        self.id = id
        self.description = description
        self.date_onborn = date_onborn
        self.gender = gender
        self.number_of_bones = number_of_bones
        self.cage = cage

    def __str__(self):
        return f"Animal(id={self.id}, description={self.description}, date_onborn={self.date_onborn}, gender={self.gender}, number_of_bones={self.number_of_bones}, cage={self.cage})"

    @abstractmethod
    def feed(self, food):
        pass

    def register(self, cage):
        print(f"Register the animal ---> {self.__class__.__name__} in cage {cage}")

    @property
    def id(self):
        return self._id

    @id.setter
    def id(self, id):
        self._id = id

    @property
    def description(self):
        return self._description

    @description.setter
    def description(self, description):
        self._description = description

    @property
    def date_onborn(self):
        return self._date_onborn

    @date_onborn.setter
    def date_onborn(self, date_onborn):
        self._date_onborn = date_onborn

    @property
    def gender(self):
        return self._gender

    @gender.setter
    def gender(self, gender):
        self._gender = gender

    @property
    def number_of_bones(self):
        return self._number_of_bones

    @number_of_bones.setter
    def number_of_bones(self, number_of_bones):
        self._number_of_bones = number_of_bones

    @property
    def cage(self):
        return self._cage

    @cage.setter
    def cage(self, cage):
        self._cage = cage
